cpp/src/toppra/toppra.hpp

```diff
@@ -2,11 +2,19 @@
 #define TOPPRA_TOPPRA_HPP
 
 #include <memory>
+#include <iostream>
 
 #include <Eigen/Core>
 #include <Eigen/StdVector>
 #include <vector>
 
+#ifdef TOPPRA_DEBUG_ON
+#define TOPPRA_LOG_DEBUG(X) std::cout << "[DEBUG]: " <<  X << std::endl
+#else
+#define TOPPRA_LOG_DEBUG(X) if (true) {} else {std::cout << "[DEBUG]: " <<  X << std::endl;}
+#endif
```

Comment:
Here you can choose between
```cpp
#ifdef TOPPRA_DEBUG_ON
#define TOPPRA_LOG_DEBUG(X) std::cout << "[DEBUG]: " <<  X << std::endl
#else
#define TOPPRA_LOG_DEBUG(X) ((void)0) // this is just to avoid compiler warnings.
#endif
```
or
```cpp
#ifdef TOPPRA_DEBUG_ON
#define TOPPRA_LOG_DEBUG std::cout << "[DEBUG]: "
#else
#define TOPPRA_LOG_DEBUG if (true) {} else std::cout << "[DEBUG]: "
#endif
```

The two versions are used as
```cpp
TOPPRA_LOG_DEBUG("my message" << my_variable); // version 1
TOPPRA_LOG_DEBUG << "my message" << my_variable << std::endl; // version 2
```

Reply:
Thanks for the suggestion. I will go with the first one since there are quite a few debug messages in this form already.